File: fathersDay/Dsa/Day 25 (2108. Find First Palindromic String in the Array)/Solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

class Solution {
public:
    bool isPalindrome(const std::string& s) {
        std::string reversed_s = s;
        std::reverse(reversed_s.begin(), reversed_s.end());
        return s == reversed_s;
    }

    std::string FindFirstPalindromic(std::vector<std::string>& words) {
        for (const std::string& word : words) {
            if (isPalindrome(word)) {
                return word;
            }
        }
        return "";
    }
};

int main() {
    Solution solution;
    std::vector<std::string> words = {"abc", "car", "ada", "racecar", "cool"};
    std::string result = solution.FindFirstPalindromic(words);
    std::cout << '"' << result << '"' << std::endl; // Output: "ada"
    return 0;
}

File: fathersDay/Dsa/Day 25 (2108. Find First Palindromic String in the Array)/Solution.java
class Solution {
    public boolean isPalindrome(String s) {
        return new StringBuilder(s).reverse().toString().equals(s);
    }

    public String FindFirstPalindromic(String[] words) {
        for (String word : words) {
            if (isPalindrome(word)) {
                return word;
            }
        }
        return "";
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String[] words = {"abc", "car", "ada", "racecar", "cool"};
        String result = solution.FindFirstPalindromic(words);
        System.out.println('"' + result + '"'); // Output: "ada"
    }
}
